packages/vue3/src/_dev/App.vue

```diff
@@ -1,42 +1,56 @@
 <script setup lang="ts">
 import 'uno.css';
 import '../assets/main.css';
-import { ref } from 'vue';
+import { ref, watch } from 'vue';
 import { FileInput, UploadedFile } from '~/components';
 
-const attachment = ref<File | null>(null);
+const attachments = ref<File[]>([]);
+const disabled = ref(false);
+const limit = ref(4);
 
-const uploadFile = (file: File[]) => {
-  attachment.value = file[0];
+const checkLimit = () => {
+  disabled.value = attachments.value.length >= limit.value;
 };
-const deleteFile = () => {
-  attachment.value = null;
+
+const deleteFile = (file: File) => {
+  const idx = attachments.value.indexOf(file);
+  if (idx > -1) {
+    attachments.value.splice(idx, 1);
+    checkLimit();
+  }
 };
+watch(attachments, () => {
+  checkLimit();
+});
 </script>
 
 <template>
   <div class="container">
-    <UploadedFile
-      v-if="attachment"
-      :file="attachment"
-      error="selected file is more than 2M"
-      @delete="deleteFile"
-    />
-    <div class="file-input">
-      <FileInput @input="uploadFile" />
+    <div class="files-grid">
```

Comment:
Better to name this one 
```suggestion
    <div class="files-list">
```
As `files-grid` gives the impression that it has the display property of grid